### DISCUSSION TITLE
UUID column without default in migration

### DISCUSSION THREAD

**Discussion by IlyaSemenov** [romeerez/orchid-orm#159] [Q&A] [answered]
I'm using uuid keys in the database that I'm always generating on the app side with uuidv7. As such, I don't want the database to have `DEFAULT gen_random_uuid()` to prevent mistakes (non-uuidv7 keys added by database). How do I create such columns with rake?

**Comment by romeerez** [comment, marked as answer]
The default for uuid columns is set only when using `primaryKey`, and if you don't want it you can add `default(null)`:

```ts
change(async (db) => {
  await db.createTable('tmp', (t) => ({
    pkey: t.uuid().primaryKey().default(null),
  }));
});
```

**Comment by romeerez** [comment]
It was working before but with some nuances, I published improvements for it so `.default(null)` will make rake-db ignore default instead of `DEFAULT null`, and the types for `create` for a column with `default(null)` were considering it as optional, and now it will be required for `create`.
